### DISCUSSION TITLE
[BUG]: invalid workflow file should mark PR checks as failed

### DISCUSSION THREAD

**Discussion by ShaMan123** [community/community#34408] [Actions]
## Actual: Confusing Indication
The **checks** div indicates that 1 [test](https://github.com/ShaMan123/gha-form-dropdown-options/actions/runs/3140859978/jobs/5102688773) passed as if it were the only one, though the [second test](https://github.com/ShaMan123/gha-form-dropdown-options/actions/runs/3140859976) failed to startup.

![image](https://user-images.githubusercontent.com/34343793/192689144-1f002749-2681-40dd-8054-4f6f8daefbe0.png)

![image](https://user-images.githubusercontent.com/34343793/192688984-777709fe-bf4c-4239-b11c-bf319a68bb2c.png)

### After fixing the failing workflow
![image](https://user-images.githubusercontent.com/34343793/192690297-aa6dc1dc-2c6e-4c9a-b401-b62d34c69c4a.png)

## Actions Tab
In the [actions tab](https://github.com/ShaMan123/gha-form-dropdown-options/pull/2/checks?check_run_id=8587692896) of the PR I can see the failing workflow

![image](https://user-images.githubusercontent.com/34343793/192691856-52135c04-b5e9-4a37-8cf0-6332ae617492.png)

![image](https://user-images.githubusercontent.com/34343793/192689086-b3e92870-0083-4fe1-9ab0-21b73462e704.png)

But the number of workflows in the tab is wrong, should be 2

![image](https://user-images.githubusercontent.com/34343793/192691303-0f2958a3-0203-4195-b2cc-cc0a192d6f48.png)

## Expected
Should move PR into a failing state and indicate that the second test died in the **checks** div

**Comment by DenverCoder1** [comment]
Keep in mind that not all workflows are meant for running on pull requests. Some are only for specific types of pull requests, certain branches, releases, issues, etc.

If the workflow file is invalid it means it can't be parsed so GitHub can't know when the workflow was supposed to be run.

It wouldn't make sense for an invalid "release" workflow to show up as failed on pull requests, so since the file is invalid, the best GitHub can do is not show it and tell you it is invalid when looking into why it's not running.

---

Edit: It seems from some of the screenshots you sent that for this particular invalid workflow, GitHub was able to determine it is meant to run on pull requests, probably due to an action validation error rather than a yml syntax error.

I think there are some problems that could be resolved as you described, but there are also many it would not make sense to do that with since the issue could occur anywhere in the file including in the part that decides when the workflow should run.

**Comment by ShaMan123** [comment]
Good point.
Perhaps then, what I am after is an alert showing on the repo that it contains a broken workflow

**Comment by assignUser** [comment]
If a workflow file is modified in a PR and is now invalid this should absolutely show up as a failed check or if not some other visible way. I have seen such changes happen due to a rebase/ff or similar just before someone clicks the merge button, as there is no red check it's rare that it gets noticed (especially if you have 10+ jobs in the list).
